VirtoCommerce.CatalogModule.Data/Services/CatalogSecurity.cs

```diff
@@ -0,0 +1,177 @@
+using System.Collections.Generic;
+using System.Linq;
+using VirtoCommerce.CatalogModule.Web.Security;
+using VirtoCommerce.Domain.Catalog.Model;
+using VirtoCommerce.Domain.Catalog.Services;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Security;
+
+namespace VirtoCommerce.CatalogModule.Data.Services.Security
+{
+    public class CatalogSecurity : ICatalogSecurity
+    {
+        private readonly ISecurityService _securityService;
+        private readonly ICategoryService _categoryService;
+        private readonly IPermissionScopeService _permissionScopeService;
+
+
+        public CatalogSecurity(ISecurityService securityService, ICategoryService categoryService,
+            IPermissionScopeService permissionScopeService)
+        {
+            _securityService = securityService;
+            _categoryService = categoryService;
+            _permissionScopeService = permissionScopeService;
+        }
+
+        #region ISearchCreteriaSecurity
+
+        public void ApplayUserRestrictions(SearchCriteria criteria, string userName)
+        {
+            var permissionScopes = GetReedPermissionScopes(userName);
```

Comment:
it will inflate an already large method, can it better be left a separate method?

Reply:
In that case i do not think it simplifies the method, it would be simpler to read without it. It is one line method used only once - i think it should be inlined. And variable could be named even more precisely - catalogReadPermissionScopes, as there could be other read permission scopes.